### ISSUE TITLE
java.lang.NoSuchMethodError: com.amazonaws.http.AmazonHttpClient.disableStrictHostnameVerification

### ISSUE THREAD

**Issue by myyk** [myyk/sbt-codedeploy#2]
@MojoJojo "Okay, Now I'm getting the error below. Think its conflicting with the s3 resolver plugin..."

```
[warn] two warnings found
warning: Class com.amazonaws.transform.JsonErrorUnmarshallerV2 not found - continuing with a stub.

java.lang.NoSuchMethodError: com.amazonaws.http.AmazonHttpClient.disableStrictHostnameVerification()V
        at com.amazonaws.services.s3.AmazonS3Client.init(AmazonS3Client.java:444)
        at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:391)
        at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:368)
        at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:350)
        at com.amazonaws.services.s3.AmazonS3Client.<init>(AmazonS3Client.java:334)
        at ohnosequences.ivy.S3Repository.<init>(S3Repository.java:85)
        at ohnosequences.ivy.S3Resolver.<init>(S3Resolver.java:48)
        at ohnosequences.sbt.SbtS3Resolver$autoImport$S3Resolver.<init>(SBTS3Resolver.scala:14)
        at ohnosequences.sbt.SbtS3Resolver$$anonfun$projectSettings$7$$anonfun$apply$1.apply(SBTS3Resolver.scala:101)
        at ohnosequences.sbt.SbtS3Resolver$$anonfun$projectSettings$7$$anonfun$apply$1.apply(SBTS3Resolver.scala:101)
        at ServerSettings$$anonfun$24.apply(ServerSettings.scala:125)
        at ServerSettings$$anonfun$24.apply(ServerSettings.scala:119)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.EvaluateSettings$MixedNode.evaluate0(INode.scala:175)
        at sbt.EvaluateSettings$INode.evaluate(INode.scala:135)
        at sbt.EvaluateSettings$$anonfun$sbt$EvaluateSettings$$submitEvaluate$1.apply$mcV$sp(INode.scala:69)
        at sbt.EvaluateSettings.sbt$EvaluateSettings$$run0(INode.scala:78)
        at sbt.EvaluateSettings$$anon$3.run(INode.scala:74)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
[error] java.lang.NoSuchMethodError: com.amazonaws.http.AmazonHttpClient.disableStrictHostnameVerification()V
[error] Use 'last' for the full log.
```

**Comment by myyk** [comment]
@MojoJojo what version of the "s3 resolve plugin" are you using?

**Comment by MojoJojo** [comment]
```
addSbtPlugin("ohnosequences" % "sbt-s3-resolver" % "0.14.0")
```

**Comment by myyk** [comment]
Dang, I don't know how to get around that. I think those people need to update their plugin. I created an issue. https://github.com/ohnosequences/ivy-s3-resolver/issues/13

**Comment by MojoJojo** [comment]
Thank you for your assistance so far. Think I'll revert to the older version for now as there's no getting around this issue until the S3 resolver is updated.

**Comment by myyk** [comment]
If you want to try something else out. I wonder if you can just force the version of the s3 dependency to be higher by modifying the `project/plugins.sbt` to have a like like this:

libraryDependencies += "com.amazonaws" % "aws-java-sdk-s3" % "1.10.68"

**Comment by MojoJojo** [comment]
I tried both:

```
libraryDependencies += "com.amazonaws" % "aws-java-sdk-s3" % "1.10.68"
```

and:

```
dependencyOverrides += "com.amazonaws" % "aws-java-sdk-s3" % "1.10.68"

```

but still gives me the same error.

**Comment by myyk** [comment]
Oh that's too bad. Those plugins just won't work together until they either update their aws dep versions or someone can show us some SBT magic to make that work.

**Comment by MojoJojo** [comment]
Actually, forcing it to use a newer version fixed it: 

```
libraryDependencies += "com.amazonaws" % "aws-java-sdk-s3" % "1.11.26"  force()
```

**Comment by myyk** [comment]
Oh, wow, that's great!

**Comment by MojoJojo** [comment]
Okay, here is the final fix using the latest AWS SDK:

In plugins.sbt:

```
libraryDependencies += "com.amazonaws" % "aws-java-sdk-s3" % "1.11.26"  force()
libraryDependencies += "com.amazonaws" % "aws-java-sdk-core" % "1.11.26"  force()
libraryDependencies += "com.amazonaws" % "aws-java-sdk-codedeploy" % "1.11.26"  force()
libraryDependencies += "com.amazonaws" % "aws-java-sdk-cloudformation" % "1.11.26"  force()

addSbtPlugin("ohnosequences" % "sbt-s3-resolver" % "0.14.0" excludeAll(ExclusionRule(organization = "com.amazonaws")))

addSbtPlugin("com.github.tptodorov" % "sbt-cloudformation" % "0.7.1" excludeAll(ExclusionRule(organization = "com.amazonaws")))

addSbtPlugin("com.github.myyk" % "sbt-codedeploy" % "0.5.0" excludeAll(ExclusionRule(organization = "com.amazonaws")))
```

**Comment by myyk** [comment]
👍  I guess I can close this as there's nothing really to do to this project to fix this. But this issue should help those who have this issue find a solution, I think.

**Comment by myyk** [comment]
BTW, https://github.com/ohnosequences/ivy-s3-resolver/pull/14 was merged and published along with `sbt-s3-resolver` v0.15.0.

So, you should probably be able to update your plugins, also I published an updated `sbt-codedeploy` with newer v0.6.0 with the updated aws sdk version. I'd imagine they'd work together without the changes you had to do.
